src/main.rs

```diff
@@ -119,6 +119,18 @@ fn command_build(root: String) -> Result<(), Error> {
 
     let compiled = crate::project::compile(srcs)?;
 
+    // Delete the gen directory before generating the newly compiled files
+    let gen_dir = root_path.parent().unwrap().join("gen");
```

Comment:
Can we `.map_err(|e| Error::FileIO {` for the `parent` call rather than unwrapping it? Thank you.